Ab 1

Astemir2147 opened the pull request.

AsanovAlbek committed:
Добавил стили

AsanovAlbek committed:
AndroidManifest.xml: Добавил возможность клавиатуре показываться поверх экрана
build.gradle(core): Добавил firestore
build.gradle: Обновил gradle до 7.1.2
FirestoreDataStore.kt удалён
FirestoreDataStoreImpl.kt удалён
MainActivity.kt скрыт action bar
MyDatabase.kt Обновлена версия базы
ComponentsEntity.kt Добавлен автоинкремет id
Repository.kt Изменён метод, добавлен аргумент к нему

AsanovAlbek committed:
ComponentsDao.kt Добавил добавление одного компонента
ComponentsEntity.kt Добавил автоинкремент contractId
InsertRepository.kt Добавил метод добавления компонента в бд
InsertRepositoryImpl.kt имплементировал метод
InsertInteractor.kt Добавил методы добавления в room и firebase
InsertInteractorImpl.kt имплементировал методы InsertInteractor.kt
FirebaseRepository.kt(feature_insert_data) Добавил метод для добавления компонента в firebase
FirebaseRepositoryImpl.kt имплементировал FirebaseRepository.kt
google-services.json добавил json файл для работы с firebase
InsertDataViewModel.kt Добавил новые методы из InteractorImpl.kt

AsanovAlbek committed:
InsertDataFragment.kt использовал новые методы из InsertViewModel.kt
insertModule.kt добавил в зависимости FirebaseRepository.kt(feature_insert_data), а так же добавил его в InsertInteractor
fragment_insert_data.xml сделал красивые автозаполняемые поля, ну и по мелочи навёл красоту
LoginFirebaseAuth.kt ещё больше закомментировал его)

AsanovAlbek committed:
ComponentsFragment.kt теперь в init вызывается заполнение room из firebase
ComponentsViewModel.kt добавил метод загрузки данных из firebase в room
data.kt добавил в зависимости FirebaseRepository.kt
domain.kt добавил к интерактору FirebaseRepository.kt
FirebaseRepository.kt добавил метод загрузки в room из firebase
FirebaseRepositoryImpl.kt имплементировал метод
Interactor.kt и InteractorImpl.kt добавил методы из FirebaseRepository.kt и имплементировал

Commit:
изменение в стиле на вкладке компонентов, изменена строчка с фильтами

AsanovAlbek committed:
Добавил комментарии к некоторым классам, убрал star import, убрал логи
В ComponentsFragment.kt добавил проверку на наличие сети и загрузку компонентов из Firebase
Добавил метод для удаления всех данных из бд ( метод setAllComponents в ComponentsDao.kt хоть и имеет onConflict = OnConflictStrategy.REPLACE, но почему то только добавляет данные, следовательно получается дублирование имеющихся компонентов)
В fragment_components.xml добавил id для общего LinearLayout, чтобы передать его для SnackBar
В InsertDataFragment.kt в методе openCalendar поменял формат даты с MM.dd.yyyy на dd.MM.yyyy
InsertInteractor.kt и InsertRepository.kt поменял названия переменных на более подходящие, убрал лишнюю функциональность
InsertRepositoryImpl.kt поменял статус при добавлении с Installed на ReadyForUse

AsanovAlbek committed:
Merge remote-tracking branch 'origin/AB-1' into AB-1

Commit:
исправил подключение к firebase

AsanovAlbek committed:
Удалил файлы solarlint

AsanovAlbek committed:
Добавил файлы sonarlint в игнор

AsanovAlbek committed:
InsertRepository.kt, InsertInteractor.kt, InsertDataViewModel.kt убрал метод, служивший конструктором для компонента, перенёс конструктор в Component.kt
Repository.kt, Interactor.kt, ComponentsViewModel.kt поправил метод для получения всех компонентов
InsertDataViewModel.kt поправил метод для удаления компонентов, пофиксил баг с дублированием компонентов в списках

snakelord757 commented on .idea/.gitignore:
```diff
@@ -1,3 +1,4 @@
 # Default ignored files
 /shelf/
 /workspace.xml
+sonarlint
```
Не уверен, что это сработает. Скорее всего нужно ```idea/sonarlint```

snakelord757 commented on .idea/misc.xml:
```diff
@@ -3,6 +3,23 @@
   <component name="DesignSurface">
     <option name="filePathToZoomLevelMap">
       <map>
+        <entry key="..\:/ComponentsMetric/app/src/main/res/drawable-v24/ic_calendar.xml" value="0.206" />
```
Этого файла не должно быть в ПРе

snakelord757 commented on app/google-services.json:
```diff
@@ -8,7 +8,7 @@
   "client": [
```
Этот файл надо удалить из репозитория. Приватные файлы не должны лежать в свободном доступе.
P.s. вы сохраняете эти файлы локально, но при этом здесь его не должно быть

AsanovAlbek committed:
Merge remote-tracking branch 'origin/AB-1' into AB-1

snakelord757 commented on app/src/main/java/com/example/componentsmetric/ui/MainActivity.kt:
```diff
@@ -31,6 +31,7 @@ class MainActivity : AppCompatActivity() {
         )
         setupActionBarWithNavController(navController)
         navView.setupWithNavController(navController)
+        supportActionBar?.hide()
```
Экшнбар никому не нужен?

AsanovAlbek replied:
Мы решили, что так красивее)

snakelord757 replied:
> Мы решили, что так красивее)

В таком случае у темы в конце можно добавить ```.NoActionBar```

snakelord757 commented on core/src/main/java/com/example/core/database/base/MyDatabase.kt:
```diff
@@ -1,19 +1,23 @@
  * @author Zashaev Astemir on 2022-04-09
+ *
+ * @author Asanov Albek on 2022-06-10
+ * Изменил Версию до 2, чтобы добавить автоинкремент поля componentId
```
Эти комментарии оставляйте в описании к ПР, в коде они не нужны

snakelord757 commented on core/src/main/java/com/example/core/database/dao/ComponentsDao.kt:
```diff
@@ -1,16 +1,14 @@
-import androidx.room.Insert
-import androidx.room.OnConflictStrategy
-import androidx.room.Query
+import androidx.room.*
```
Импорты через ```*``` или же ```wildcard imports``` нежелательны. Нужно импортировать только то, что тебе нужно

snakelord757 commented on feature_components/build.gradle:
```diff
@@ -39,11 +39,12 @@ dependencies {
 //    implementation files('core/libs.jar')
```
Никаких закомментированного кода в проекте быть не должно

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/data/repository/FirebaseRepositoryImpl.kt:
```diff
@@ -0,0 +1,52 @@
+    private val componentsDao: ComponentsDao
+) : FirebaseRepository{
+
+    private val firebase = FirebaseFirestore.getInstance()
```
Зависимость не должна быть в классе. В идеале написать обертку в духе ```RemoteDataSource```

snakelord757 commented on feature_components/build.gradle:
```diff
@@ -39,11 +39,12 @@ dependencies {
     def lifecycle = "2.4.1"
-
-    def room_version = "2.4.2"
+    implementation("androidx.compose.material3:material3:1.0.0-alpha13")
```
Уверены что вам нужны эти зависимости?)

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/data/repository/FirebaseRepositoryImpl.kt:
```diff
@@ -0,0 +1,52 @@
+            .addOnSuccessListener { snapshot ->
+
+                // Парсинг компонента
+                    val firebaseComponents = snapshot.documents.map { snapshotItem ->
```
Можно обернуть ```snapshotItem``` в ```with``` и обращаться напрямую без ```.``` 
```
                    val firebaseComponents = snapshot.documents.map { snapshotItem ->
                        with(snapshotItem) {
                            ComponentsEntity(
                                contractId = get("contractId") as Long,
                                componentName = get("componentName").toString(),
                                accepted = get("personWitchAccept").toString(),
                                countOfItem = get("countOfItem").toString(),
                                dateOfAccept = get("dateOfAccept").toString(),
                                statusOfComponent = get("statusOfComponent").toString()
                            )
                        }
                }
```

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/data/repository/FirebaseRepositoryImpl.kt:
```diff
@@ -0,0 +1,52 @@
+                // Парсинг компонента
+                    val firebaseComponents = snapshot.documents.map { snapshotItem ->
+                        ComponentsEntity(
+                            contractId = snapshotItem.get("contractId") as Long,
```
Если глянешь код ```DocumentSnapshot```, то можешь найти там функции, которые сразу вернут типизированное значение, например ```getLong```, ```getString``` и т.д.

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/data/repository/FirebaseRepositoryImpl.kt:
```diff
@@ -0,0 +1,52 @@
+                    val firebaseComponents = snapshot.documents.map { snapshotItem ->
+                        ComponentsEntity(
+                            contractId = snapshotItem.get("contractId") as Long,
+                            componentName = snapshotItem.get("componentName").toString(),
```
Нельзя использовать строковые константы в явном виде. Сделай здесь же ```companion object``` и вынеси туда все константы с именами типа ```COMPONENT_NAME_FIELD```

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/data/repository/FirebaseRepositoryImpl.kt:
```diff
@@ -0,0 +1,52 @@
+                        )
+                }
+
+                    GlobalScope.launch {
```
Функция должна вызываться в самом верхнем скоупе, т.е. скоупе ViewModel. ```GlobalScope``` нельзя использовать

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/data/repository/FirebaseRepositoryImpl.kt:
```diff
@@ -0,0 +1,52 @@
+                }
+
+                    GlobalScope.launch {
+                        withContext(Dispatchers.IO) {
```
Диспетчеры тоже нельзя использовать в явном виде. Нужно завести какой-нибудь ```DispatcherProvider``` и использовать как зависимость (пригодится когда будете писать тесты). Переключение контекста тоже должно происходить не в репозитории

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/data/repository/FirebaseRepositoryImpl.kt:
```diff
@@ -0,0 +1,52 @@
+                        }
+                    }
+            }
+                // Если получен не успешно ( пока ничего не делаем )
```
Тогда и нет смысла держать пустой коллбек

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/data/repository/RepositoryImpl.kt:
```diff
@@ -1,50 +1,31 @@
 import com.example.feature_components.data.model.Component
 import com.example.feature_components.domain.Repository
 
 class RepositoryImpl(private val dao: ComponentsDao) : Repository {
```
Назвать интерфейс и реализацию более осмысленно

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/data/repository/RepositoryImpl.kt:
```diff
@@ -1,50 +1,31 @@
-    override suspend fun getAllComponentFromDb(status: String): List<Component> {
-        return dao.getComponentsForStatus(status).map { it.componentsEntityToComponents() }
+    override suspend fun getAllComponentFromDb(): List<Component> {
+        return dao.getAllComponents().map { it.componentsEntityToComponents() }
```
Маппинг делаем в интеракторах/юзкейсах

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/domain/FirebaseRepository.kt:
```diff
@@ -0,0 +1,14 @@
+ * @author Asanov Albek 25.06.2022
+ */
+
+interface FirebaseRepository {
```
```FirebaseRepository``` тоже такое себе название, лучше что-нибудь более приближенное к тому, что оно делает, типа ```RemoteComponentsRepository```

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/domain/Repository.kt:
```diff
@@ -12,11 +13,15 @@ interface Repository {
-    /** Занести данные в бд */
-    suspend fun setAllComponentToDb()
+    /** Занести данные в бд
+     * upd : Asanov Albek 13.06.2022
```
Поправь доку

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/domain/interactor/Interactor.kt:
```diff
@@ -9,15 +10,18 @@ import com.example.feature_components.data.model.Component
 interface Interactor {
 
-    /** Получение всех контрактов из БД */
+    /** Получение всех контрактов из БД с определенным статусом [status]*/
```
Если здесь же перечислишь возможные ```status```, то будет круто

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/domain/interactor/Interactor.kt:
```diff
@@ -9,15 +10,18 @@ import com.example.feature_components.data.model.Component
     suspend fun searchComponent(query: String): List<Component>
+
+    /** Добавление в room компонентов, полученных из firebase
+     * @author Asanov Albek */
```
Авторство может быть у классов, top-level функций, но не у функции в классе/интерфейсе

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/domain/interactor/Interactor.kt:
```diff
@@ -9,15 +10,18 @@ import com.example.feature_components.data.model.Component
+     * @author Asanov Albek */
+    suspend fun getComponentsFromFirebase()
+
+    suspend fun deleteRoomComponents(components : List<Component>)
```
Не хватает доки

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/domain/interactor/Interactor.kt:
```diff
@@ -1,5 +1,6 @@
 package com.example.feature_components.domain.interactor
```
Более осмысленное название для интерфейса и реализации

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/presentation/ComponentsFragment.kt:
```diff
@@ -1,20 +1,20 @@
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
 import android.os.Bundle
+import android.util.Log
```
Неиспользуемый импорт

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/presentation/ComponentsFragment.kt:
```diff
@@ -23,16 +23,32 @@ class ComponentsFragment : Fragment(R.layout.fragment_components), SearchView.On
+    }
+
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
         componentsBinding = FragmentComponentsBinding.inflate(inflater, container, false)
```
Когда ты закидываешь лейаут в конструктор, то можно уже не переопределять метод ```onCreateView```. Биндинг можешь подцепить в ```onViewCreated```. В этом методе в качестве параметра приходит ```view```, который надо будет прокидывать в ```FragmentComponentsBinding.bind```

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/presentation/ComponentsFragment.kt:
```diff
@@ -23,16 +23,32 @@ class ComponentsFragment : Fragment(R.layout.fragment_components), SearchView.On
+    /** Метод, проверяющий наличие подключения к сети, и исходя из этого
+     * подгружает данные из firebase в room, или же сообщает об ошибке */
+    private fun upsertComponents() {
+
```
Лишний пробел

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/presentation/ComponentsFragment.kt:
```diff
@@ -23,16 +23,32 @@ class ComponentsFragment : Fragment(R.layout.fragment_components), SearchView.On
+     * подгружает данные из firebase в room, или же сообщает об ошибке */
+    private fun upsertComponents() {
+
+        if (isOnline()) {
```
Лучше будет ```isNetworkAvailable```

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/presentation/ComponentsFragment.kt:
```diff
@@ -23,16 +23,32 @@ class ComponentsFragment : Fragment(R.layout.fragment_components), SearchView.On
+            componentsViewModel.init()
+        }
+        else {
+            snackBar("Нет подключения к интернету")
```
Вынести в ```strings```

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/presentation/ComponentsFragment.kt:
```diff
@@ -77,8 +93,37 @@ class ComponentsFragment : Fragment(R.layout.fragment_components), SearchView.On
 
-    fun searchComponent(query: String) {
+    private fun snackBar(message : String) {
+        val view = requireActivity().findViewById<View>(android.R.id.content)
```
Заменяется на ```requireView()```

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/presentation/ComponentsFragment.kt:
```diff
@@ -77,8 +93,37 @@ class ComponentsFragment : Fragment(R.layout.fragment_components), SearchView.On
+    }
+
+    private fun searchComponent(query: String) {
         val searchQuery = "%$query%"
```
Никаких взаимодействий с чем либо внутри фрагментов/активити. Вынеси во ViewModel

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/presentation/ComponentsFragment.kt:
```diff
@@ -77,8 +93,37 @@ class ComponentsFragment : Fragment(R.layout.fragment_components), SearchView.On
+     * Функция проверки подключения к интернету (wifi или мобильная связь)
+     */
+
+    fun isOnline(): Boolean {
```
Функция должна быть приватной

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/presentation/ComponentsFragment.kt:
```diff
@@ -77,8 +93,37 @@ class ComponentsFragment : Fragment(R.layout.fragment_components), SearchView.On
+    fun isOnline(): Boolean {
+        val connectivityManager =
+            requireActivity().getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+        if (connectivityManager != null) {
```
```connectivityManager``` никогда не будет null. Как минимум по тому, что ты не предусмотрел nullability поля, В случае если здесь будет null ты упадешь с NullPointerException (не указав что поле может быть null ты берешь ответственность за содержимое этого поля)

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/presentation/ComponentsFragment.kt:
```diff
@@ -77,8 +93,37 @@ class ComponentsFragment : Fragment(R.layout.fragment_components), SearchView.On
+            val capabilities =
+                connectivityManager.getNetworkCapabilities(connectivityManager.activeNetwork)
+            if (capabilities != null) {
+                if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
```
Если у тебя есть  ```if-else if-else``` то вероятнее всего можно упростить условие, например:
```
val hasCellularConnection = capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)
val hasWiFiConnection = capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)
val hasEthernetConnection = capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)
return hasCellularConnection || hasWiFiConnection || hasEthernetConnection
```

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/presentation/ComponentsFragment.kt:
```diff
@@ -77,8 +93,37 @@ class ComponentsFragment : Fragment(R.layout.fragment_components), SearchView.On
+     * Функция проверки подключения к интернету (wifi или мобильная связь)
+     */
+
+    fun isOnline(): Boolean {
```
Это тоже не должно происходить в фрагменте. Можно сделать делегат, который будет ответственным за проверку сети

AsanovAlbek replied:
А будет ли правильно сделать use case, отвечающий за это?

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/presentation/ComponentsViewModel.kt:
```diff
@@ -39,8 +41,18 @@ class ComponentsViewModel(
+
     fun init() {
-        setContractsToDb()
+        deleteAllContractsToDb()
```
Зачем удалять содержимое таблицы при каждой инициализации ViewModel?

snakelord757 commented on feature_components/src/main/java/com/example/feature_components/presentation/ComponentsViewModel.kt:
```diff
@@ -56,19 +68,11 @@ class ComponentsViewModel(
             withContext(Dispatchers.IO) {
                 val contracts = componentsInteractor
-                    .getComponentsFromDb(DatabaseConstStatus.READY_FOR_USE)
+                    .getComponentsForStatus(DatabaseConstStatus.READY_FOR_USE)
```
Лучше не затаскивать константы из базы данных. Просто продублируй их здесь в ```companion object```

snakelord757 commented on feature_components/src/main/res/layout/fragment_components.xml:
```diff
@@ -1,53 +1,47 @@
-            android:gravity="center"
-            android:text="@string/received" />
+        xmlns:android="http://schemas.android.com/apk/res/android"
+        android:backgroundTint="#DFE4D6"
```
Цвет вынести в ```colors```

snakelord757 commented on feature_components/src/main/res/layout/fragment_components.xml:
```diff
@@ -1,53 +1,47 @@
 <?xml version="1.0" encoding="utf-8"?>
```
Ужасное форматирование. Прожми ```Ctrl+Alt+L```

snakelord757 commented on feature_components/src/main/res/layout/fragment_components.xml:
```diff
@@ -1,53 +1,47 @@
-            android:layout_weight="1"
-            android:gravity="center"
-            android:text="@string/received" />
+        xmlns:android="http://schemas.android.com/apk/res/android"
```
Ты уже добавил пространство имён во второй строке

snakelord757 commented on feature_components/src/main/res/values/styles.xml:
```diff
@@ -11,4 +11,13 @@
     </style>
 
+    <style name="Widget.App.Button.OutlinedButton.IconOnly" parent="Widget.Material3.Button.OutlinedButton">
+        <item name="android:textSize">15sp</item>
```
Размер текста в ```dimens```

snakelord757 commented on feature_components/src/main/res/values/styles.xml:
```diff
@@ -11,4 +11,13 @@
+        <item name="android:textSize">15sp</item>
+        <item name="backgroundTint">@android:color/transparent</item>
+        <item name="strokeColor">@color/gray</item>
+        <item name="strokeWidth">1dp</item>
```
Размер границ в ```dimens```

snakelord757 commented on feature_components/src/main/res/values/styles.xml:
```diff
@@ -11,4 +11,13 @@
+        <item name="backgroundTint">@android:color/transparent</item>
+        <item name="strokeColor">@color/gray</item>
+        <item name="strokeWidth">1dp</item>
+        <item name="rippleColor">#DFE4D6</item>
```
Цвета в ```colors```

snakelord757 commented on feature_insert_data/build.gradle:
```diff
@@ -39,6 +40,11 @@ dependencies {
 //    implementation 'androidx.appcompat:appcompat:1.4.1'
```
Комментарии нужно удалить

snakelord757 commented on feature_components/src/main/res/layout/fragment_components.xml:
```diff
@@ -1,53 +1,47 @@
+    </com.google.android.material.button.MaterialButtonToggleGroup>
+
 
     <androidx.recyclerview.widget.RecyclerView
```
Мне кажется есть смысл заменить рутовый лейаут на ```ConstraintLayout```, т.к. позиционирование сломанно. Список при скролле до конца не отображается
<img src="https://user-images.githubusercontent.com/14038823/179046997-de2621c7-e27c-448b-8402-89b08bf5cbc0.png" height=700/>

snakelord757 commented on feature_insert_data/src/main/java/com/example/feature_insert_data/data/models/Component.kt:
```diff
@@ -0,0 +1,38 @@
+    val personWitchAccept: String,
+    val countOfItem: String,
+    val dateOfAccept: String,
+    val statusOfComponent: String
```
```statusOfComponent``` имеет ограниченный перечень состояний, поэтому надо бы его сделать enum`ом

snakelord757 commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,4 +1,4 @@
-#Thu Mar 10 22:15:23 MSK 2022
```
Этого файла не должно быть в ПР

snakelord757 commented on feature_login/src/main/java/com/example/feature_login/data/base/LoginFirebaseAuth.kt:
```diff
@@ -1,34 +1,34 @@
-    //             }
-    //         }
-    // }
+     /*fun signIn(user: AuthDateUser) {
```
Удалить закомментированный код

snakelord757 commented on feature_insert_data/src/main/res/values/strings.xml:
```diff
@@ -1,4 +1,33 @@
+        <item>Процессоры</item>
+        <item>Видеокарты</item>
+    </string-array>
+    <string-array name="brands">
```
В отличии от категорий бренды и модели меняются чаще, поэтому стоит подумать как получать их динамически

snakelord757 commented on feature_insert_data/src/main/java/com/example/feature_insert_data/presentation/validator/InsertValidator.kt:
```diff
@@ -0,0 +1,30 @@
+package com.example.feature_insert_data.presentation.validator
+
+import java.text.SimpleDateFormat
+import java.util.*
```
wildcard import

snakelord757 commented on feature_insert_data/src/main/java/com/example/feature_insert_data/presentation/InsertDataFragment.kt:
```diff
@@ -1,21 +1,168 @@
+
+        val dateSetListener = DatePickerDialog.OnDateSetListener {
+                _, year, month, day ->
+            //"${month+1}.$day.$year"
```
Удалить код с комментариями

snakelord757 commented on feature_insert_data/src/main/java/com/example/feature_insert_data/presentation/InsertDataFragment.kt:
```diff
@@ -1,21 +1,168 @@
+                binding.clickableDateTv.text = date
+            }
+            else {
+                snackBar("Некорректная дата")
```
Строки в ```strings```

snakelord757 commented on feature_insert_data/src/main/java/com/example/feature_insert_data/presentation/InsertDataFragment.kt:
```diff
@@ -1,21 +1,168 @@
+        }
+    }
+
+    private fun snackBar(message : String) {
```
Не надо дублировать её везде, лучше сделать функцию-расширение для фрагментов

snakelord757 commented on feature_insert_data/src/main/java/com/example/feature_insert_data/presentation/InsertDataFragment.kt:
```diff
@@ -1,21 +1,168 @@
+        binding.addComponentButton.setOnClickListener {
+
+            if (isAnyEmptyField()) {
+                snackBar("Пожалуйста, заполните все поля")
```
Строки в ```strings```

snakelord757 commented on feature_insert_data/src/main/java/com/example/feature_insert_data/presentation/InsertDataFragment.kt:
```diff
@@ -1,21 +1,168 @@
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
         homePageBinding = FragmentInsertDataBinding.inflate(inflater, container, false)
```
Та же ошибка, что и в фрагменте компонентов

snakelord757 commented on feature_insert_data/src/main/java/com/example/feature_insert_data/presentation/InsertDataFragment.kt:
```diff
@@ -1,21 +1,168 @@
 
+import android.app.DatePickerDialog
 import android.os.Bundle
+import android.util.Log
```
Неиспользуемый импорт

snakelord757 commented on feature_insert_data/src/main/java/com/example/feature_insert_data/domain/InsertRepository.kt:
```diff
@@ -0,0 +1,36 @@
+ * @author Asanov Albek 25.06.2022
+ */
+
+interface InsertRepository {
```
Нужно более осмысленное название

snakelord757 commented on feature_insert_data/src/main/java/com/example/feature_insert_data/data/repository/InsertRepositoryImpl.kt:
```diff
@@ -0,0 +1,44 @@
+    override fun getComponentName(
+        category: String,
+        brand: String,
+        model: String) = StringBuilder().apply {
```
Заменяется одной строкой
```
"$category $brand $model "
```

snakelord757 commented on feature_insert_data/src/main/java/com/example/feature_insert_data/data/repository/FirebaseRepositoryImpl.kt:
```diff
@@ -0,0 +1,46 @@
+                "dateOfAccept" to component.dateOfAccept,
+                "statusOfComponent" to component.statusOfComponent
+            )
+        ).addOnSuccessListener { }.addOnFailureListener { }
```
Если в коллбеках ничего не происходит, то и не нужно их добавлять

snakelord757 commented on feature_insert_data/src/main/java/com/example/feature_insert_data/data/repository/FirebaseRepositoryImpl.kt:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    // Экземпляр базы
+    val firebase = FirebaseFirestore.getInstance()
```
Про зависимость писал выше

snakelord757 commented on feature_insert_data/src/main/java/com/example/feature_insert_data/data/extantion/Extantion.kt:
```diff
@@ -0,0 +1,13 @@
+
+import com.example.core.database.entity.ComponentsEntity
+import com.example.feature_insert_data.data.models.Component
+/** Данный из presentation в модель для бд */
```
Данный кем?)

snakelord757 reviewed (changes requested):
Так же необходимо поправить форматирование кода. Много лишних пробелов, неправильных переносов и т.д.

AsanovAlbek committed:
Правки:
build.gradle в feature_components: Убрал закомментированный код

feature_components:
Перенёс объявление binding в onViewCreated и объявил через метод bind
Вынес проверку сети из фрагмента во вью модель
Перенёс snackBar в Extantion.kt и сделал extantion функцией для фрагмента
Вынес строковые значения статусов в константы в ComponentsViewModel.kt
Переименовал репозитории и интерактор
Вынес некоторые значения в dimens.xml, colors.xml и strings.xml (значения в styles.xml в том числе)
убрал лишние коллбеки
Исправил форматирование, Ctrl+Alt+L

feature_insert_data:
Переименовал репозитории и интерактор
Вынес snackBar в Extantion.kt
Убрал лишние коллбеки
Вынес сообщения для snackBar в strings.xml
Убрал закоментированный код

Прочее:
Убрал star import
Исправил доку

Добавлено:
CallbackDataFromFirebase.kt - интерфейс для получения данных из firebase
NetworkDelegate.kt - делегат для работы с сетью
Добавил NetworkDelegate.kt к зависимостям и имплементировал в ComponentsViewModel.kt
